<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Motor Control System</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <!-- Left Panel: Controls -->
     <widget class="QWidget" name="leftPanel" native="true">
      <property name="maximumWidth">
       <number>300</number>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <!-- Connection Group -->
       <item>
        <widget class="QGroupBox" name="connectionGroupBox">
         <property name="title">
          <string>Connection</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="portLabel">
            <property name="text">
             <string>Port:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="portComboBox">
            <item><property name="text"><string>COM5</string></property></item>
            <item><property name="text"><string>COM6</string></property></item>
            <item><property name="text"><string>/dev/ttyACM0</string></property></item>
            <item><property name="text"><string>/dev/ttyUSB0</string></property></item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="baudLabel">
            <property name="text">
             <string>Baud:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="baudComboBox">
            <item><property name="text"><string>9600</string></property></item>
            <item><property name="text"><string>115200</string></property></item>
            <item><property name="text"><string>921600</string></property></item>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="connectButton">
            <property name="text">
             <string>Connect</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QPushButton" name="disconnectButton">
            <property name="text">
             <string>Disconnect</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       
       <!-- Motion Setup Group -->
       <item>
        <widget class="QGroupBox" name="motionGroupBox">
         <property name="title">
          <string>Motion Setup</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="stepsLabel">
            <property name="text">
             <string>Steps:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="stepsSpinBox">
            <property name="minimum">
             <number>-10000</number>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="value">
             <number>1000</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="velocityLabel">
            <property name="text">
             <string>Max Velocity:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="velocitySpinBox">
            <property name="maximum">
             <number>2000</number>
            </property>
            <property name="value">
             <number>500</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="accelLabel">
            <property name="text">
             <string>Acceleration:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="accelSpinBox">
            <property name="maximum">
             <number>5000</number>
            </property>
            <property name="value">
             <number>1000</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QPushButton" name="planMotionButton">
            <property name="text">
             <string>Plan Motion</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       
       <!-- PID Tuning Group -->
       <item>
        <widget class="QGroupBox" name="pidGroupBox">
         <property name="title">
          <string>PID Tuning</string>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="kpLabel">
            <property name="text">
             <string>P:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="kpSpinBox">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <double>100.00</double>
            </property>
            <property name="singleStep">
             <double>0.10</double>
            </property>
            <property name="value">
             <double>1.50</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="kiLabel">
            <property name="text">
             <string>I:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="kiSpinBox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="maximum">
             <double>10.00</double>
            </property>
            <property name="singleStep">
             <double>0.01</double>
            </property>
            <property name="value">
             <double>0.100</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="kdLabel">
            <property name="text">
             <string>D:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="kdSpinBox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="maximum">
             <double>10.00</double>
            </property>
            <property name="singleStep">
             <double>0.01</double>
            </property>
            <property name="value">
             <double>0.050</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="kfLabel">
            <property name="text">
             <string>F:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="kfSpinBox">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <double>10.00</double>
            </property>
            <property name="singleStep">
             <double>0.10</double>
            </property>
            <property name="value">
             <double>0.80</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QPushButton" name="applyGainsButton">
            <property name="text">
             <string>Apply Gains</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       
       <!-- Control Group -->
       <item>
        <widget class="QGroupBox" name="controlGroupBox">
         <property name="title">
          <string>Control</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="startButton">
            <property name="text">
             <string>‚ñ∂ Start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stopButton">
            <property name="text">
             <string>‚ñ† Stop</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="estopButton">
            <property name="text">
             <string>‚èπ E-STOP</string>
            </property>
            <property name="styleSheet">
             <string>background-color: red; color: white;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="homeButton">
            <property name="text">
             <string>üè† Home</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       
       <!-- Recording Group -->
       <item>
        <widget class="QGroupBox" name="recordingGroupBox">
         <property name="title">
          <string>Recording</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QPushButton" name="recordButton">
            <property name="text">
             <string>‚óè Record</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="saveRunButton">
            <property name="text">
             <string>Save Run</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="loadRunButton">
            <property name="text">
             <string>Load Run</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    
    <item>
     <!-- Right Panel: Plots and Console -->
     <widget class="QWidget" name="rightPanel" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <!-- Position Plot -->
       <item>
        <widget class="QCustomPlot" name="positionPlot" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>250</height>
          </size>
         </property>
        </widget>
       </item>
       
       <!-- Velocity Plot -->
       <item>
        <widget class="QCustomPlot" name="velocityPlot" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>250</height>
          </size>
         </property>
        </widget>
       </item>
       
       <!-- Current Values -->
       <item>
        <widget class="QGroupBox" name="currentGroupBox">
         <property name="title">
          <string>Current Values</string>
         </property>
         <property name="maximumHeight">
          <number>80</number>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Position:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="currentPosLabel">
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="styleSheet">
             <string>font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Velocity:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="currentVelLabel">
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="styleSheet">
             <string>font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Error:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="currentErrorLabel">
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="styleSheet">
             <string>font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Phase:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="currentPhaseLabel">
            <property name="text">
             <string>IDLE</string>
            </property>
            <property name="styleSheet">
             <string>font-weight: bold;</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       
       <!-- Console -->
       <item>
        <widget class="QGroupBox" name="consoleGroupBox">
         <property name="title">
          <string>Console Log</string>
         </property>
         <property name="maximumHeight">
          <number>150</number>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QTextEdit" name="consoleTextEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="styleSheet">
             <string>font-family: 'Courier New'; font-size: 10pt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>

